numba/core/unsafe/refcount.py

```diff
@@ -37,6 +37,10 @@ def codegen(context, builder, signature, args):
                 # "%zu" is not portable.  just truncate refcount to 32-bit.
                 # that's good enough for a debugging util.
                 refct_32bit = builder.trunc(refct, ir.IntType(32))
+                refct_32bit = builder.sub(
```

Comment:
Given that this is a debugging tool, I don't think it's a good idea to make adjustment to the actual refct.

Reply:
Since the argument gets casted on the way in when `get_refcount` is called the refcount is now one bigger than it used to be. This could even be considered a feature as this brings the behaviour in line with `sys.getrefcount` which is counting the reference held by itself as well. 

Hence, subtracting from the refcount should definitely go before this gets merged. This is was just intended as a tactical solution to avoid tempering with the testcases until everything else works.